File: features/merchant/review_inbox/src/main/java/com/tokopedia/review/feature/inbox/buyerreview/data/mapper/ShopFavoritedMapper.java
package com.tokopedia.review.feature.inbox.buyerreview.data.mapper;

import com.tokopedia.review.feature.inbox.buyerreview.data.pojo.FavoriteCheckResult;
import com.tokopedia.review.feature.inbox.buyerreview.domain.model.inboxdetail.CheckShopFavoriteDomain;
import com.tokopedia.review.feature.inbox.buyerreview.network.ErrorMessageException;

import retrofit2.Response;
import rx.functions.Func1;

/**
 * @author by nisie on 9/26/17.
 */

public class ShopFavoritedMapper implements Func1<Response<FavoriteCheckResult>, CheckShopFavoriteDomain> {

    @Override
    public CheckShopFavoriteDomain call(Response<FavoriteCheckResult> response) {
        if (response.isSuccessful()) {
            if (response.body().getShopIds() != null) {
                return mappingToDomain(!response.body().getShopIds().isEmpty());
            } else {
                throw new ErrorMessageException("");
            }
        } else {
            throw new RuntimeException(String.valueOf(response.code()));
        }
    }

    private CheckShopFavoriteDomain mappingToDomain(boolean isFavorited) {
        return new CheckShopFavoriteDomain(isFavorited);
    }
}

File: features/merchant/review_inbox/src/main/java/com/tokopedia/review/feature/inbox/buyerreview/data/mapper/ShopFavoritedMapper.kt
package com.tokopedia.review.feature.inbox.buyerreview.data.mapper

import com.tokopedia.review.feature.inbox.buyerreview.data.pojo.FavoriteCheckResult
import com.tokopedia.review.feature.inbox.buyerreview.domain.model.inboxdetail.CheckShopFavoriteDomain
import retrofit2.Response
import rx.functions.Func1
import javax.inject.Inject

/**
 * @author by nisie on 9/26/17.
 */
class ShopFavoritedMapper @Inject constructor() : Func1<Response<FavoriteCheckResult?>?, CheckShopFavoriteDomain> {

    override fun call(response: Response<FavoriteCheckResult?>?): CheckShopFavoriteDomain {
        return if (response?.isSuccessful == true) {
            mappingToDomain(response.body()?.shopIds?.isNullOrEmpty() == true)
        } else {
            throw RuntimeException(response?.code().toString())
        }
    }
    private fun mappingToDomain(isFavorited: Boolean): CheckShopFavoriteDomain {
        return CheckShopFavoriteDomain(isFavorited)
    }
}